config/validate/util/highlight.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2015 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"fmt"
+	"strings"
+)
+
+func Highlight(srcb []byte, offset int64) (int, int, string) {
+	src := string(srcb)
+
+	startOfLine := strings.LastIndex(src[:offset], "\n") + 1
```

Comment:
Should probably cap offset to `len(src)` at the start of this function.

Reply:
I'm not sure if I agree. The options are:
1) Make it have weird behavior by capping offset to `len(src)-1` (`offset` is an index; we'd want the `-1`). This could hide bugs where offset is being generated wrong.
2) Also return an error. This means checking that everywhere and it's not exactly actionable.
3) Die (what we're currently doing)
4) Die explicitly where we check offset and panic with a better message than the default one

I'm much more of a fan of 3/4. 2 is just a pain and 1 could hide bugs in the validation code.

Reply:
Sure, 4 sounds good to me. (Though I'm fine with this as is as well.)

Reply:
I think I'll leave it as is since it should be pretty obvious from the `index out of bounds` message what's going on.